MM2RandoLib/Random/MT19937Seed.cs

```diff
@@ -0,0 +1,176 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using MM2Randomizer.Extensions;
+using Troschuetz.Random.Generators;
+
+namespace MM2Randomizer.Random
+{
+    /// <summary>
+    /// Provides basic randomization methods
+    /// </summary>
+    public sealed class MT19937Seed : ISeed
+    {
+        //
+        // Constructors
+        //
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("CodeQuality", "IDE0051:Remove unused private members", Justification = "Private constructors are used by the SeedFactory.")]
+        private MT19937Seed(String in_SeedString)
+        {
+            if (null == in_SeedString)
+            {
+                throw new ArgumentNullException(nameof(in_SeedString));
+            }
+
+            UInt32[] seedArray = in_SeedString.Select(x => Convert.ToUInt32(x)).ToArray();
```

Comment:
Yes, that is what this is doing.  The constructor to the MT19937Generator() takes an array of UInt32 as the "seed," so we use the Convert class to convert each character in the string to an array.  It really isn't inefficient, especially if we want to use any kind of string as a potential seed.  I think adding more hashing algorithms is a great idea, and we can have a debug UI to test different ones.

Reply:
I'm not at all concerned about efficiency. My comment about suboptimality has to do with coverage of the value space. If you're converting each ascii character to a UInt32 of the entropy, there is a chance that the RNG ends up biased. My thought was to convert it into a more dense representation and then convert than to UInt32 chunks that you pass in.

Reply:
Ah, now I understand.  I don't think this actually makes a difference, because the value space is actually an infinitely large array of UInt32, so this should cover the fact that each UInt32 is likely not being utilized.  But, I don't know how the array "seed" affects the randomization.  If we wanted to improve this a little without getting too complicated, we could combine every two characters before converting, since unicode characters are UInt16, and the array is made up of UInt32.

Reply:
What about just taking the FNV hash for now and using that value? I feel like that should be the simplest change for now. Then we can explore alternatives later.